pyproject.toml

```diff
@@ -1,10 +1,106 @@
+build-backend = 'mesonpy'
 requires = [
-    "wheel",
-    "setuptools",
-    "Cython>=0.29.31"
+    'meson-python>=0.11',
+    "meson>=1.1",
+    'ninja',
+    'wheel',
+    'oldest-supported-numpy',
+    'Cython>=0.29.31',
+    'pyproject-metadata>=0.5.0',
+    "tomli>=1.0.0; python_version < '3.12'",
+    'h5py',
+    'fabio'
 ]
-build-backend = "setuptools.build_meta"
+
+
+[project.optional-dependencies]
+full = [ 'pyopencl; platform_machine in "i386, x86_64, AMD64"',  # For silx.opencl
+          'Mako',                                                # For pyopencl reduction
+          'qtconsole',                                            # For silx.gui.console
+          'matplotlib >= 3.1.0',                                 # For silx.gui.plot
+          'PyOpenGL',                                            # For silx.gui.plot3d
+          'python-dateutil',                                     # For silx.gui.plot
+          'scipy',                                               # For silx.math.fit demo, silx.image.sift demo, silx.image.sift.test
+          'pooch',                                               # For scipy.datasets.ascent
+          'Pillow',                                              # For silx.opencl.image.test
+          'pint',                                                # For silx.io.dictdump
+          'PySide6 >= 6.4'                                       # PyQt6, PyQt5  # For silx.gui
+        ]
+test = ['build',                     # To build the project
+        'wheel',                     # To build wheels
+        'Sphinx',                    # To build the documentation in doc/
+        'sphinx-autodoc-typehints',  # For leveraging Python type hints from Sphinx
+        'sphinx-design',             # For tabs and grid in documentation
+        'pillow',                    # For loading images in documentation generation
+        'pydata_sphinx_theme',       # Sphinx theme
+        'nbsphinx',                  # For converting ipynb in documentation
+        'pandoc',                    # For documentation Qt snapshot updates
+        'black[jupyter]',            # Formatting
+        'pytest',                    # For testing
+        'pytest-xvfb',               # For GUI testing
+        'pytest-cov',                # For coverage
+        'pytest-mock',
+        'hdf5plugin',                # For HDF5 compression filters handling
+        'pyinstaller>=6.0.0; sys_platform == "win32"',
+        ]
+opencl = [ "pyopencl", "Mako" ]
+
+[project.urls]
+homepage = 'http://www.silx.org'
+documentation = 'http://www.silx.org/doc/silx/latest/'
+source = 'https://github.com/silx-kit/silx'
+download = 'https://github.com/silx-kit/silx/releases'
+tracker = 'https://github.com/silx-kit/silx/issues'
+
+
+[project.scripts]
+silx = 'silx.__main__:main'
+
+
+[tool.cibuildwheel]
+# Skip 32-bit builds and PyPy
+skip = ["*-win32", "*-manylinux_i686", "pp*", "*musllinux*"]
+
 
 [tool.black]
 required-version = 25
```

Comment:
This could be a string

Reply:
Maybe it could, but it is better to have it in a structured way to handle it programmatically from `tomli`